models/spring-ai-openai/src/main/java/org/springframework/ai/openai/metadata/support/OpenAiResponseHeaderExtractor.java

```diff
@@ -71,7 +71,7 @@ public static RateLimit extractAiResponseHeaders(ResponseEntity<?> response) {
 
 	private static Duration getHeaderAsDuration(ResponseEntity<?> response, String headerName) {
 		var headers = response.getHeaders();
-		if (headers.containsKey(headerName)) {
+		if (null != headers.getFirst(headerName)) {
```

Comment:
This can be optimized a bit since at the moment the headers are fetched twice. The `getFirst` is not needed. The entire section could be 

```java
var values = headers.get(headerName);
if (!CollectionUtils.isEmpty(values)) {
    return DurationFormatter.TIME_UNIT.parse(values.get(0));
}
```

or

```java
var value = headers.getFirst(headerName);
if (value != null) {
    return DurationFormatter.TIME_UNIT.parse(value);
}
```